src/tictacclojure/player.clj

```diff
@@ -1,28 +1,37 @@
 (ns tictacclojure.player
   (:require [tictacclojure.board :as board]
+            [tictacclojure.console-ui :as ui]
+            [tictacclojure.text-prompts :as prompts]
             [tictacclojure.minimax :as minimax]))
 
-(defn- console-in
-  []
-  (read-line))
+(defn- validate-text-move
+  [board text-position]
+  (try
+    (let [position (Integer/parseInt text-position)]
+      (board/position-available? board position))
+    (catch NumberFormatException exception false)))
 
 (defn- pick-move-for-human
-  [mark]
-  [(Integer/parseInt (console-in)) mark])
+  [{:keys [board]} mark]
+  (let [position (Integer/parseInt
```

Comment:
Makes sense, though I am not sure how I could avoid it here, as I need to first check that the user's input string can be parsed into an Integer, and then do the conversion..Ideally I would use `Integer/canParseToInt` in the validation and then `Integer/parseInt` when actually parsing, but I don't think Java offers such facility?